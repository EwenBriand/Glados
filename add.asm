section .data
    ; no data yet
section .text
    global _start

_start:
    xor rax, rax
    mov rax, 1
    push rax
    xor rax, rax
    mov rax, 2
    pop rdi
    add rax, rdi
    mov rdi, 60 ; sys_exit
    xor rax, rax
    syscall
