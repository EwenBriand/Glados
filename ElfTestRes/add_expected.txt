
ElfTestRes/mov.o:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <_start>:
   0:	01 d8                	add    %ebx,%eax
   2:	01 d1                	add    %edx,%ecx
   4:	01 fe                	add    %edi,%esi

0000000000000006 <_imm>:
   6:	83 c0 01             	add    $0x1,%eax
   9:	83 c3 2a             	add    $0x2a,%ebx
   c:	83 c1 f6             	add    $0xfffffff6,%ecx
